### PR TITLE
Task/inba 280 preject reducer mapping

### PR DESCRIPTION
**Pull request by Perry5**
#### What's this PR do?
Builds and returns a composite object for the front end
#### Related JIRA tickets:
[280](https://jira.amida-tech.com/browse/INBA-280)
#### How should this be manually tested?
With the Indaba backend set up, try hitting this end point with Postman
`http://localhost:3005/test/v0.2/projects/aggregate`
#### Screenshots (if appropriate):
screen shot 2017-08-09 at 7 14 51 pm

### CODE REVIEW

File: backend/app/controllers/projects.js [hunk 1]
```diff
@@ -182,6 +303,8 @@ module.exports = {
         });
     }
 
+
+
```
Comment (Perry5) [inline backend/app/controllers/projects.js]: Need to take out white spaces

File: backend/config.js [hunk 2]
```diff
@@ -23,10 +23,10 @@ var environments = {
             expiresAfterSeconds: 360000 * 24 // 24 hour
         },
         pgConnect: {
-            user: process.env.RDS_USERNAME || process.env.INDABA_PG_USERNAME || 'db_user',
+            user: process.env.RDS_USERNAME || process.env.INDABA_PG_USERNAME || 'indabauser',
             testuser: process.env.RDS_TESTUSER || process.env.INDABA_PG_TESTUSER || 'test', // make trust method for this user in PostgreSQL Client Authentication Configuration File (pg_hba.conf)
             password: process.env.RDS_PASSWORD || process.env.INDABA_PG_PASSWORD || 'password',
-            database: process.env.INDABA_PG_DB || 'database',
+            database: 'indabatestuser' || process.env.INDABA_PG_DB || 'indaba',
```
Comment (jmfadeley) [inline backend/config.js]: Remove `indatatestuser` and just reference the `env` file.

File: backend/config.js [hunk 3]
```diff
@@ -111,11 +111,11 @@ var environments = {
             expiresAfterSeconds: 360000 * 24 // 24 hour
         },
         pgConnect: {
-            user: process.env.RDS_USERNAME || process.env.INDABA_PG_USERNAME || 'db_user',
+            user: process.env.RDS_USERNAME || process.env.INDABA_PG_USERNAME || 'indabauser',
             // make trust method for this user in PostgreSQL Client Authentication Configuration File (pg_hba.conf)
             testuser: process.env.RDS_TESTUSER || process.env.INDABA_PG_TESTUSER || 'test',
             password: process.env.RDS_PASSWORD || process.env.INDABA_PG_PASSWORD || 'password',
-            database: process.env.INDABA_PG_DB || 'database',
+            database: 'indabatestuser' || process.env.INDABA_PG_DB || 'indaba',
```
Comment (jmfadeley) [inline backend/config.js]: Ditto as above.

File: backend/app/router.js [hunk 4]
```diff
@@ -82,6 +82,9 @@ router.route('/:realm/v0.2/projects/:id/products')
 router.route('/:realm/v0.2/projects/:id/surveys')
     .get(authenticate('token').always, projects.surveyList);
 
+router.route('/:realm/v0.2/projects/aggregate')
```
Comment (jmfadeley) [inline backend/app/router.js]: Good name.

**Comment by jmfadeley** [comment]
Okay so needs two changes thus far:
The `users` array should just be an array of numbers of the ids, and not an array of objects. With lodash I think you can just go `_.map(users, 'id')`.

For `userGroups`, we need the `groupName` and array for the `users` in that group:

`    userGroups: [
        {
            id: 11,
            name: 'Researchers',
            users: [41, 25, 22, 31],
        }, {
            id: 13,
            name: 'Managers',
            users: [13, 71],
        }],
`

**Comment by jmfadeley** [comment]
Last things we need:

- `subjects` should be mapped from the `ProductUOA`, not from `UserUOA`. 
- Right now, we're setting it up simply as one product to one project to one survey. `productIDs` should be a single `productID` and just return the project's associated product identity number.

**Comment by Perry5** [comment]
Pushed new updated to PR. Please review and let me know if there is anything else we need for this.

**Pull request merged**